source/material_model/utilities.cc

```diff
@@ -1280,6 +1280,46 @@ namespace aspect
                                                                   n_phase_transitions_per_composition,
                                                                   true);
       }
+
+
+      std::vector<double>
+      parse_map_differences_to_values(const std::vector<double> input_values,
+                                      const std::unique_ptr<std::vector<unsigned int>> &values_per_key,
+                                      const PhaseUtilities::PhaseAveragingOperation operation)
+      {
+        unsigned field_index = 0;
+        std::vector<double> return_values;
+        for (const unsigned int &n_values: *values_per_key)
+          {
+            for (unsigned int n_value=0; n_value<n_values; n_value++)
+              {
+                if (n_value == 0)
+                  {
+                    // push back the value from the first phase of a composition
+                    return_values.push_back(input_values[field_index]);
+                  }
+                else
+                  {
+                    // handle values of the other phases of a compostion
+                    if (operation == PhaseUtilities::logarithmic)
+                      {
+                        // in case of logarithmic, the values are added by the logarithmics
```

Comment:
```suggestion
                        // in case of logarithmic addition, the difference values are added to the logarithmic of the previous phase value.
```